queryscript/src/compile/external.rs

```diff
@@ -0,0 +1,71 @@
+use crate::{
+    ast::{Ident, Located, SourceLocation},
+    types::{AtomicType, Type},
+};
+use snafu::prelude::*;
+
+use super::{
+    error::{Result, RuntimeSnafu},
+    inference::mkcref,
+    schema::{CRef, CTypedExpr, Expr, MType},
+    CompileError,
+};
+
+// XXX If a record has two fields with the same name, we should throw an error. Eventually,
+// we should support this, because SQL engines do.
```

Comment:
This isn't really true -- tables, views and subqueries can't have duplicate columns, only top-level SELECT queries can.  We'll only have to support this if we enable injecting a SQL query into the connection string or table definition somehow, which I'd argue is a pretty anti-QS thing to do.

Reply:
Ah interesting, let me update the comment. I guess it will affect unsafe expressions but not tables, the latter being much more important for us.